src/components/FileUpload.tsx

```diff
@@ -54,16 +55,105 @@ export default function FileUpload() {
         setErrorOpen(false);
     };
 
+    async function getToken() {
+        const token = await firebase.auth().currentUser?.getIdToken();
+
+        if (token === undefined) return {};
+
+        return token;
+    }
+
     useEffect(() => {
         globalThis.uploadedDatasets = datasets;
     }, [datasets]);
 
+    async function uploadDatatset(key: string) {
+        if (key === undefined) {
+            enqueueSnackbar("Something went wrong", {
+                variant: "error"
+            });
+            return;
+        }
+
+        fetch(process.env.NEXT_PUBLIC_GQL_API_UPDATED, {
+            method: "POST",
+            headers: {
+                "content-type": "application/json",
+                Authorization: (await getToken()) as unknown as string
+            },
+            body: JSON.stringify({
+                query: `mutation addDataset($datasetKey: String!){
+                         addDataSet(dataSetKey:$datasetKey){
+                            id
+                            name
+                            datasets
+                        }
+                }
+                `,
+                variables: {
+                    datasetKey: `${process.env.NEXT_PUBLIC_DATASET_API}/${key}`
+                }
+            })
+        })
+            .then((resp) => resp.json())
```

Comment:
You should await the fetch instead of using `.then`, you're in an async function. You can still use `.catch` too